File: backend/app/schemas/boletin.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class BoletinBase(BaseModel):
    tipo_documento: str | None = None
    consecutivo: int | None = None
    fecha: datetime | None = None
    modulo: str | None = None
    opcion: str | None = None
    impacto: str | None = None
    categoria: str | None = None
    con_documentacion: bool = False
    asunto: str | None = None
    clase_documento: str | None = None
    advertencia: str | None = None
    instructivo_descripcion: str | None = None
    mes: int
    anio: int
    archivo: str | None = None


class BoletinCreate(BoletinBase):
    pass


class BoletinResponse(BoletinBase):
    oid: int
    fecha_registro: datetime

    model_config = ConfigDict(from_attributes=True)


class BoletinPeriodo(BaseModel):
    mes: int
    anio: int


class BoletinImportResponse(BaseModel):
    mes: int
    anio: int
    inserted_rows: int

File: frontend/src/types/boletin.ts
export interface ApiBoletin {
  oid: number;
  tipo_documento: string | null;
  consecutivo: number | null;
  fecha: string | null;
  modulo: string | null;
  opcion: string | null;
  impacto: string | null;
  categoria: string | null;
  con_documentacion: boolean;
  asunto: string | null;
  clase_documento: string | null;
  advertencia: string | null;
  instructivo_descripcion: string | null;
  mes: number;
  anio: number;
  archivo: string | null;
}
export interface ApiBoletinPeriodo { mes: number; anio: number; }
export interface ApiBoletinImportResult { mes: number; anio: number; inserted_rows: number; }
